pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -482,6 +485,43 @@ class ModelParser implements Parser {
         return r
     }
 
+    @Nonnull ModelASTAxes parseAxes(Statement stmt) {
+        //Probably want to support axis1 = ['a','b','c'] rather than axis1('a','b','c')
+        def r = new ModelASTAxes(stmt)
+
+        def m = matchBlockStatement(stmt)
+        if (m==null) {
+            errorCollector.error(r, Messages.ModelParser_ExpectedBlockFor("axes"))
+            return r
+        } else {
+            eachStatement(m.body.code) { s ->
+                def mc = matchMethodCall(s)
+                if (mc == null) {
+                    // Not sure of a better way to deal with this - it's a full-on parse-time failure.
+                    errorCollector.error(r, Messages.ModelParser_ExpectedAxes())
+                } else {
+                    def axsKey = parseKey(mc.method)
+
+                    List<Expression> args = ((TupleExpression) mc.arguments).expressions
+                    if (args.isEmpty()) {
+                        errorCollector.error(axsKey, Messages.ModelParser_NoArgForAxis(axsKey.key))
+                    } else {
+                        r.axes[axsKey] = []
+
+                        args.each { a ->
+                            if (!(a instanceof ConstantExpression) && !(a instanceof GStringExpression)) {
```

Comment:
So this brings up an interesting question: do we want to be able to support other expressions in `axes`? I think we probably will - at the very least, we're going to want some way of specifying multiple axis values as a parameter (though probably not with [Extended Choice plugin](https://wiki.jenkins.io/display/JENKINS/Extended+Choice+Parameter+plugin) because it's kinda problematic), which would probably be referred to with something like `axisX(params.SOME_PARAM_NAME)`, which would end up being a list in practice. And we might well want to support letting users call a shared library function to get the axis list...

Hmm. As I think about it, maybe something like this as a syntax?
```
axes {
  axis(name: "LITERAL_AXIS", values: [1, 2, 3])
  axis(name: "PARAM_AXIS", values: params.SOME_MULTI_PARAM)
  axis(name: "FUNCTION_AXIS", values: someSharedLibMethod())
}
```

Your thoughts? =)

Reply:
I would think. I do like things declarative so calling out and not having repeatable. I suppose that argument has been had before.